lib/kcerts/ssh.go

```diff
@@ -99,6 +99,31 @@ func (a SSHAgent) Valid() bool {
 	return err == nil
 }
 
+// Principals returns a map where the keys are the CA's PKS and the certs identities are the values
+func (a SSHAgent) Principals() (map[string][]string, error) {
+	conn, err := net.Dial("unix", a.Socket)
+	if err != nil {
+		return nil, err
+	}
+	defer conn.Close()
+	keys, err := agent.NewClient(conn).List()
+	if err != nil {
+		return nil, err
+	}
+	toReturn := map[string][]string{}
```

Comment:
so, maps in go are randomized (for good reasons). If I had to guess, if you are printing this out anywhere you're getting a different order every time. My recommendation would be to use a plain list.

I would also recommend a small struct to reduce risk of errors. Eg, something like:
```
type KnownCert struct {
  MD5 string
  Principal []string
}

toReturn := []KnownCert{}
```

Reply:
(this would also show the results in the same order as the ssh-agent, which is a bouns)

Reply:
Implemented, nice.